lib/resty/healthcheck.lua

```diff
@@ -222,6 +224,121 @@ local ngx_timer_at do
 end
 
 
+local run_locked
+do
+  local resty_lock = require "resty.lock"
```

Comment:
Any particular reason why resty_lock is not `require`'d at the top of the file, like the rest of the dependencies?

Reply:
Yes! Aside from code organization and encapsulation, the biggest reason is to create intentional friction against change. Ideally we keep our locking code very simple and do not allow it to intermingle with higher level business logic, which changes more often and increases the chance of introducing a bug. Sequestering `resty.lock` to this small scope should hopefully make future maintainers think twice about putting code which uses locks outside of this block, and a commit which moves `resty.lock` back up to the outer scope should therefore be looked at with close scrutiny.

Reply:
I see. In general I like to have all dependencies listed at the top because that way the top of the file acts like a "parameter definition" for the file - you look at the top and see all the "dependencies". Adding `require`s in other places makes them less obvious - in order to know what a file depends on, you have to "hunt" or "grep" through the file, which is less convenient.

>  hopefully make future maintainers think twice about putting code which uses locks outside of this block

I think that is a good reason. Let's help them by adding a couple comments to document this, instead of hoping that they take the hint. At the top we could put:

```
...
local ssl = require("ngx.ssl")
-- local resty_lock = require "resty.lock" -- required later in the file
```

So that the dependency is more easily visible in the customary place.

And here we could add a comment:
```
do
  -- resty_lock is restricted to this scope on purpose.
  -- If you need it outside of this scope, you need a very good reason first (see https://github.com/Kong/lua-resty-healthcheck/pull/112 )
  local resty_lock = require "resty.lock"
```

Reply:
Good points! I will update the file.